### ISSUE TITLE
Bug in sigma computation / zero probabilities?

### ISSUE THREAD

**Issue by dmacd** [alonfnt/tsnex#19]
I'm not a tSNE expert but it seems to not be working still...the embeddings are not being updated.

My test code (wrapper, uses random init and default learning rate for both methods internally)
```
dataset_5d = clustered.ClusteredDataset(n_clusters=3, overlap=0.1, dimension=5,
                                samples_per_cluster=100, seed=42, scale=1)

clustered.plot_clusters_tsne(dataset_5d,
                             perplexity=10,
                             n_components=2,
                             n_iter=1000,
                             use_tsnex=<False or True>) 
```

tSNEX:

![Image](https://github.com/user-attachments/assets/7af11fde-4ec2-44b7-a31b-0c4e38fb3b08)

sklearn.manifold TSNE

![Image](https://github.com/user-attachments/assets/9da60b86-5486-443d-a4c1-cb4cdf8ae1e2)

Added some prints in the body of the `tsnex.transform`:
```
distances=[[0.         0.13794167 0.04009846 ... 5.6287184  5.43171    5.9593706 ]
 [0.13794167 0.         0.19798449 ... 4.2292767  4.1751423  4.5052915 ]
 [0.04009846 0.19798449 0.         ... 5.9484954  5.793983   6.2988844 ]
 ...
 [5.6287184  4.2292767  5.9484954  ... 0.         0.09234023 0.05120108]
 [5.43171    4.1751423  5.793983   ... 0.09234023 0.         0.16975167]
 [5.9593706  4.5052915  6.2988844  ... 0.05120108 0.16975167 0.        ]]
P=[[0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]
 ...
 [0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]]
sigma=0.00390625
```

**Comment by alonfnt** [comment]
Hey @dmacd, thank you a lot for this!! I've tried to fix it in #20 but before I merge it, I'd appreciate if you could test it in your problem to avoid releasing broken code (again 😅). 

To check it, just install directly from the branch
```
pip install git+https://github.com/alonfnt/tsnex.git@19-bug-in-sigma-computation-zero-probabilities
```
Really appreciate the feedback!

**Comment by dmacd** [comment]
Awesome and thank you for putting this together, looks like its working now!

(Sorry for the delay, internet went out for a day 🤷‍♂)
